instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/ui/routeline/RouteLineLayersTest.kt

```diff
@@ -543,6 +548,40 @@ class RouteLineLayersTest : BaseTest<BasicNavigationViewActivity>(
         )
     }
 
+    @OptIn(ExperimentalPreviewMapboxNavigationAPI::class)
+    @Test
+    fun lineDepthOcclusionFactorIsAppliedForRouteLine() = sdkTest {
+        val factor = 0.85
+        val options =
+            MapboxRouteLineOptions.Builder(activity).lineDepthOcclusionFactor(factor).build()
+        val routeLineApi = MapboxRouteLineApi(options)
+        val routeLineView = MapboxRouteLineView(options)
+        val style = activity.mapboxMap.getStyle()!!
+        val result = routeLineApi.setNavigationRoutes(
+            listOf(
+                route1,
+                route2,
+                route3,
+            )
+        )
+        routeLineView.renderRouteDrawData(style, result)
+
+        delay(1000) // use callback when NAVAND-1148 lands
+        val layers = style.styleLayers.mapNotNull { style.getLayer(it.id) }
+        val lineLayers = layers.filterIsInstance(LineLayer::class.java)
+        assertTrue(lineLayers.size >= 3) // 3 routes
```

Comment:
```suggestion
val lineLayers = withTimeoutOrNull(10000) {
            var lineLayers: List<LineLayer>
            do {
                delay(50)
                lineLayers = style.styleLayers.mapNotNull { style.getLayer(it.id) }
                    .filterIsInstance(LineLayer::class.java)
            } while (lineLayers.size < 3) // 3 routes
            lineLayers
        } ?: error("layer weren't initialised")
```
Maybe wait until layers are initialised to avoid flakiness on slow devices?

Reply:
The 1000 delay is used in the tests above. They are not flaky. Anyway, there soon will be a callback.
But yeah, I can use your option as well. I'll only use it here though, I won't touch other tests in this class.